pkg/reconciler/pipelinerun/resources/pipelinerunstate.go

```diff
@@ -336,6 +314,31 @@ func (facts *PipelineRunFacts) checkFinalTasksDone() bool {
 	return facts.checkTasksDone(facts.FinalTasksGraph)
 }
 
+// getPipelineTasksCount returns the count of successful tasks, failed tasks, cancelled tasks, skipped task, and incomplete tasks
+func (facts *PipelineRunFacts) getPipelineTasksCount() (int, int, int, int, int) {
```

Comment:
We may want to use a struct here instead of 5 return values of the same type (it can be unexported so it would be a implementation detail)

Reply:
yup debated to use `struct` vs such function 🤔  I see favor of `struct` 👍

Reply:
Done! @vdemeester PTAL 🙏